subsys/net/l2/wifi/Kconfig

```diff
@@ -27,3 +27,20 @@ config WIFI_MGMT_RAW_SCAN_RESULTS_ONLY
 	  to the application.
 
 endif # WIFI_MGMT_RAW_SCAN_RESULTS
+
+config WIFI_NATIVE
```

Comment:
IMO (but it's not only my opinion) the "wifi_native" nomenclature is a bit confusing. Especially "native Wi-Fi stack"- it suggests that we have a complete Wi-FI stack (including MAC) for Zephyr. In reality though, if I understood the intent of this PR correctly, it's more about providing a framework to register a Wi-Fi supplicant instance, that can be shared among multiple interfaces.

Reply:
Agreed, even I was thinking the same after publishing this PR, but thought will wait for others to chime in. Should we just call this `supplicant` support? but it might be mistaken with `wpa_supplicant`.